### ISSUE TITLE
Wrong line numbers using assertEquals when helper method declared in seperate file

### ISSUE THREAD

**Issue by AugustNagro** [scalameta/munit#855]
When using a MUnit test inside a [helper function](https://scalameta.org/munit/docs/tests.html#declare-tests-inside-a-helper-function) that lives in a different file than the FunSuite subtype, the error output is very unhelpful.

I've made a minimal reproducer here: https://github.com/AugustNagro/munit-helper-bad-trace

```
MySuite:
+ this works 0.005s
  ==> X MySuite.this fails  0.021s munit.ComparisonFailException: /home/augustnagro/prog/test-projects/munit-helper-bad-trace/src/test/scala/MySuite.scala:4
  3:class MySuite extends munit.FunSuite {
  4:  mySharedTest(this, "Postgres")
  5:}
  values are not the same
  => Obtained
  abc
  => Diff (- obtained, + expected)
  -abc
  +abd
  at munit.Assertions.failComparison(Assertions.scala:278)
  [error] Failed: Total 2, Failed 1, Errors 0, Passed 1
  [error] Failed tests:
  [error] 	MySuite
  [error] (Test / test) sbt.TestsFailedException: Tests unsuccessful
  [error] Total time: 1 s, completed Nov 26, 2024, 12:09:19 AM
```

If assertEquals printed the stacktrace (like assert() and as suggested in https://github.com/scalameta/munit/issues/782 ), that would make things much better.

**Comment by tgodzik** [comment]
I've been wondering whether hiding the stack trace is that useful. I think it was done to avoid having traces from munit, but this behaviour is flaky as seen here.

**Comment by AugustNagro** [comment]
Yeah I also understand that I'm trying to do something against the grain here. If it were configurable at least that would be great.

**Comment by tgodzik** [comment]
Ok, it looks like it would never print the stack trace and that was on purpose, fixing it in https://github.com/scalameta/munit/pull/856

Should help with your case too or at least shed more light on it

**Comment by tgodzik** [comment]
The release is running, let me know if that helps https://github.com/scalameta/munit/actions/runs/12119500567

**Comment by AugustNagro** [comment]
Apologies for the delay. I can confirm that assertEquals now shows the stack trace. Thank you very much @tgodzik !
